lib/tools/parse-id3.js

```diff
@@ -0,0 +1,250 @@
+      frame.value = parseUtf8(frame.data, 1, frame.data.length).replace(/\0*$/, '');
+      // text information frames supports multiple strings, stored as a terminator separated list [ID3v2.4.0 section 4.2.]
+      frame.values = frame.value.split('\0');
+    },
+    'TXXX': function(frame) {
+      var descriptionEndIndex;
+
+      if (frame.data[0] !== textEncodingDescriptionByte.Utf8) {
+        // ignore frames with unrecognized character encodings
+        return;
+      }
+
+      descriptionEndIndex = typedArrayIndexOf(frame.data, 0, 1);
+
+      if (descriptionEndIndex === -1) {
+        return;
+      }
+
+      // parse the text fields
+      frame.description = parseUtf8(frame.data, 1, descriptionEndIndex);
+      // do not include the null terminator in the tag value
+      // frames that allow different types of encoding contain terminated text
+      // [ID3v2.4.0 section 4.]
+      frame.value = parseUtf8(
+        frame.data,
+        descriptionEndIndex + 1,
+        frame.data.length
+      ).replace(/\0*$/, '');
+      frame.data = frame.value;
+    },
+    'W*': function(frame) {
+      // parse URL field; URL fields are always represented as ISO-8859-1 [ID3v2.4.0 section 4.]
+      // if the value is followed by a string termination all the following information should be ignored [ID3v2.4.0 section 4.3]
+      frame.url = parseIso88591(frame.data, 0, frame.data.length).replace(/\0.*$/, '');
+    },
+    'WXXX': function(frame) {
+      var descriptionEndIndex;
+
+      if (frame.data[0] !== textEncodingDescriptionByte.Utf8) {
+        // ignore frames with unrecognized character encodings
+        return;
+      }
+
+      descriptionEndIndex = typedArrayIndexOf(frame.data, 0, 1);
+
+      if (descriptionEndIndex === -1) {
+        return;
+      }
+
+      // parse the description and URL fields
+      frame.description = parseUtf8(frame.data, 1, descriptionEndIndex);
+      // URL fields are always represented as ISO-8859-1 [ID3v2.4.0 section 4.]
+      // if the value is followed by a string termination all the following information
+      // should be ignored [ID3v2.4.0 section 4.3]
+      frame.url = parseIso88591(
+        frame.data,
+        descriptionEndIndex + 1,
+        frame.data.length
+      ).replace(/\0.*$/, '');
+    },
+    'PRIV': function(frame) {
+      var i;
+
+      for (i = 0; i < frame.data.length; i++) {
+        if (frame.data[i] === 0) {
+          // parse the description and URL fields
+          frame.owner = parseIso88591(frame.data, 0, i);
+          break;
+        }
+      }
+      frame.privateData = frame.data.subarray(i + 1);
+      frame.data = frame.privateData;
+    }
+  };
+
+// TODO add dispatch type parsing
+var parseId3Frames = function(data) {
+  var frameSize, frameHeader,
+    frameStart = 10,
+    tagSize = 0,
+    frames = [];
+
+  // if not valid id3 return early.
+  if (data.length < 10 ||
```

Comment:
It might be useful here to give a bit more of an explanation to why it isn't valid. Specifically why the length can't be less than 10.

Reply:
I guess this is header length in case if data is Uint8

Reply:
I'll add a bit more of an explanation here.